modules/renter/siafile/metadata.go

```diff
@@ -0,0 +1,199 @@
+package siafile
+
+import (
+	"math"
+	"os"
+	"time"
+
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+type (
+	// Metadata is the metadata of a SiaFile and is JSON encoded.
+	Metadata struct {
+		staticVersion   [16]byte          // version of the sia file format used
+		staticFileSize  int64             // total size of the file
+		staticMasterKey crypto.TwofishKey // masterkey used to encrypt pieces
+		staticPieceSize uint64            // size of a single piece of the file
```

Comment:
We need 2 more fields for encryption, the first field needs to indicate the type of encryption, and the second field needs to be a backup field

Reply:
*a backup key. Siafiles will have 2 keys, one that you give out when you share the file and one that you don't give out. The second key prevents hosts from deduplicating files that you've shared, because they can't decrypt the pieces you didn't give out the second key for.

Then I think at the chunk level we'll need a byte indicating what piece index starts using the backup key instead of the master key for that chunk.

Reply:
Does it make sense to have partial file sharing?